src/sentry/static/sentry/app/views/performance/realUserMonitoring/transactionVitals.tsx

```diff
@@ -21,69 +21,132 @@ type Props = {
-    const max = decodeScalar(location.query.endMeasurements);
+  renderVitalGroup(vitals: WebVital[], summaryResults: any, colors: string[]) {
+    const {location, organization, eventView, dataFilter} = this.props;
 
     return (
-      <DiscoverQuery
+      <MeasurementsHistogramQuery
         location={location}
         orgSlug={organization.slug}
         eventView={eventView}
-        limit={1}
+        numBuckets={NUM_BUCKETS}
+        measurements={vitals.map(vital => WEB_VITAL_DETAILS[vital].slug)}
+        dataFilter={dataFilter}
       >
-        {summaryResults => {
+        {multiHistogramResults => {
+          const isLoading = summaryResults.isLoading || multiHistogramResults.isLoading;
+          const error =
+            summaryResults.error !== null || multiHistogramResults.error !== null;
           return (
-            <Panel>
-              <MeasurementsHistogramQuery
-                location={location}
-                orgSlug={organization.slug}
-                eventView={eventView}
-                numBuckets={NUM_BUCKETS}
-                measurements={vitals.map(vital => WEB_VITAL_DETAILS[vital].slug)}
-                min={min}
-                max={max}
-                dataFilter={dataFilter}
-              >
-                {results => (
-                  <React.Fragment>
-                    {vitals.map((vital, index) => {
-                      const error =
-                        summaryResults.error !== null || results.error !== null;
-                      const alias = getAggregateAlias(
-                        `percentile(${vital}, ${PERCENTILE})`
-                      );
-                      const summary =
-                        summaryResults.tableData?.data?.[0]?.[alias] ?? null;
-                      return (
-                        <VitalCard
-                          key={vital}
-                          location={location}
-                          isLoading={summaryResults.isLoading || results.isLoading}
-                          error={error}
-                          vital={WEB_VITAL_DETAILS[vital]}
-                          summary={summary as number | null}
-                          chartData={results.histograms?.[vital] ?? []}
-                          colors={[colors[index]]}
-                          eventView={eventView}
-                          organization={organization}
-                          min={min}
-                          max={max}
-                        />
-                      );
-                    })}
+            <React.Fragment>
+              {vitals.map((vital, index) => {
+                const alias = getAggregateAlias(`percentile(${vital}, ${PERCENTILE})`);
+                const summary = summaryResults.tableData?.data?.[0]?.[alias] ?? null;
+                const vitalSlug = WEB_VITAL_DETAILS[vital].slug;
+
+                return (
+                  <React.Fragment key={vital}>
```

Comment:
Do you need a fragment here? you could add a key to the results of `renderVitalCard`.

Reply:
I could, I just thought it was more explicit having the key where the call to `.map(...)` is rather than in the `renderVitalCard` method which only returns a single `ReactNode`.